src/core/lib/iomgr/socket_utils_common_posix.cc

```diff
@@ -267,9 +267,13 @@ static bool g_default_server_tcp_user_timeout_enabled = true;
 #endif
 #define SOCKET_SUPPORTS_TCP_USER_TIMEOUT_DEFAULT 0
 #else
-// For non-Linux, TCP_USER_TIMEOUT won't be used.
+// For non-Linux, TCP_USER_TIMEOUT will be used if TCP_USER_TIMEOUT is defined.
+#ifdef TCP_USER_TIMEOUT
+#define SOCKET_SUPPORTS_TCP_USER_TIMEOUT_DEFAULT 0
+#else
 #define TCP_USER_TIMEOUT 0
```

Comment:
Do we need to define TCP_USER_TIMEOUT here at all?

Reply:
Code below uses `TCP_USER_TIMEOUT` no matter this is provided by the platform. So it has to be defined anyway.

Reply:
It sounds wrong when you have both macros TCP_UESR_TIMEOUT and SOCKET_SUPPORTS_TCP_UESR_TIMEOUT_DEFAULT... Can the code just refer to the second macro (I assume it is our own?)